modules/route-table/main.tf

```diff
@@ -33,7 +33,7 @@ resource "aws_ec2_transit_gateway_route_table_propagation" "this" {
 ################################################################################
 
 resource "aws_ec2_transit_gateway_route" "this" {
-  for_each = { for k, v in var.routes : k => v if var.create }
+  for_each = { for route in var.static_routes : route.destination_cidr_block => route if var.create }
```

Comment:
We still use `for_each` to prevent changes, it's just the parameter that it's a list instead of a map. We use the `destination_cidr_block` as the key (it's a good candidate because duplicated cidrs are not allowed in the routes).

I think this simplifies using the module but if that's not a good Terraform practice we can stick with maps